### ISSUE TITLE
history-graph: wrong auto-refresh (input_boolean)

### ISSUE THREAD

**Issue by ildar170975** [home-assistant/frontend#15498]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Continuation of this problem: https://github.com/home-assistant/frontend/issues/13128
Currently I observe the problem for non-numerical entities only.

Test card:

![image](https://user-images.githubusercontent.com/71872483/219959541-7b7f24ed-c16a-4555-98e3-3984d4936b21.png)

```
type: vertical-stack
cards:
  - type: entities
    entities:
      - entity: input_boolean.flag
        secondary_info: last-changed
  - type: history-graph
    entities:
      - input_boolean.flag
    hours_to_show: 1
    refresh_interval: 0
```

Toggle the entity:
![image](https://user-images.githubusercontent.com/71872483/219959562-f39c986f-37a4-49f0-a79c-34e31da565e5.png)

The graph is updated not immediately, after ~1 minute:
![image](https://user-images.githubusercontent.com/71872483/219959580-f000ff95-b66b-4964-aeb4-f4067de52f57.png)

### Describe the behavior you expected

The graph is updated immediately.

### Steps to reproduce the issue

as above

### What version of Home Assistant Core has the issue?

2023.2.5

### In which browser are you experiencing the issue with?

Chrome 110.0.5481.97 

### Which operating system are you using to run this browser?

Win10x64

**Comment by karwosts** [comment]
I'm not understanding what you say the problem is. What does it mean to be "updated" immediately when you click the button? 

If the graph "updates" instantly, then the new state will be essentially 0 time or 0 pixels wide, so it would not even be visible on the graph. What do you expect to happen exactly?

I observed a boolean changing in both directions and I didn't see anything that seemed wrong to me. Both graphs seem to update at roughly once per minute, which I don't think is unreasonable?

![graph_progress_on_to_off](https://user-images.githubusercontent.com/32912880/219963852-29dafd81-0de1-4cac-9d35-58c414319a4a.gif)

![graph_progress_off_to_on](https://user-images.githubusercontent.com/32912880/219963858-2a9a2d2b-e861-4f1c-b644-bf0c9ace474f.gif)

**Comment by ildar170975** [comment]
> What does it mean to be "updated" immediately when you click the button?

Agree, this was not a correct statement from my side.
Right after toggling the new state is ~1px, which is OK.
But it keeps staying this "1px" for a minute.
If you press F5 - then the graph is updated and the new state is much bigger than "1px".
So, the problem is that this update does not happen "soon" (not "immediately"), it happens after ~1 minute.

**Comment by karwosts** [comment]
I can't speak to the tradeoffs of performance for increasing the refresh interval to faster than 1 minute. I question if it really makes that much difference, if you have large number of hours to show, if you refresh it 10 seconds after changing, the bar may be 2px wide instead of 1px, not really a significant notable difference. 

If what you're saying is it's hard to see the current state, maybe it would be nice if there was a little bubble/circle to the right of the chart that just prominently displayed the color of the current state, to make it obvious to see recent changes, but I'm not sure if that would be useful or just distracting. 

And interestingly enough it looks like line (numeric) charts have a much slower refresh interval than timeline (state). They seem to be enforcing a 5 minute refresh interval to redraw the X axis that is not enforced for timeline charts. But I do agree that it is easier to notice instant changes on the line chart, due to the large vertical line. Not sure if that's a bug or not. 

https://github.com/home-assistant/frontend/blob/dev/src/components/chart/state-history-chart-line.ts#L149
https://github.com/home-assistant/frontend/blob/dev/src/components/chart/state-history-chart-timeline.ts#L67

Timeline redraws anytime "endTime" changes, which seems frequent, but Line chart does not.

**Comment by ildar170975** [comment]
Thank you for a detailed research.
Let me explain an origin of this issue.

Since I started working with HA, I am registering questions/tasks/problems in some document (kind of registry).
I am revising old issues, closing them if / doing additional research if needed.

Last year I noticed that history-graphs stopped updating.
Some issue were registered here for this problem.
Today I revised my old records for this problem; an issue for a numerical graph is solved; and for a non-numerical graph I see a **comparingly significant delay**.
This delay is NOT critical.
As you noticed in a real life with `hours_to_show: 24` the "new-state-bar" will be thin and not vivid even after 1 minute.
Let's say that my issue is for a "perfection".
